internal/controller/postgrescluster/cluster_test.go

```diff
@@ -753,7 +754,8 @@ ownerReferences:
 	`))
 	assert.Assert(t, marshalMatches(service.Spec, `
 ports:
-- name: postgres
+- appProtocol: tcp
+  name: postgres
```

Comment:
❓ Is `tcp` the correct value here? The Kubernetes [docs](https://kubernetes.io/docs/concepts/services-networking/service/#application-protocol) and API [reference](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#ServiceSpec) seem to say these should be IANA service names. Based on that guidance I expect `appProtocol: postgresql` here. Would Istio behave correctly in that case?

📝 I've opened #3437 to add IANA constants in two packages.